### ISSUE TITLE
Index Templates don't include analysis settings

### ISSUE THREAD

**Issue by tstallings** [elastic/elasticsearch-dsl-py#1029]
I was just starting to adapt some code to use index templating rather than trying to create the index if it doesn't exist, and when I take an existing Index and call `as_template` it doesn't add the custom analyzers I've defined to the settings field. The result of not adding the custom analyzers is that calls to `save` the template fail because the analyzers aren't present. Not sure if this was intentional or not, but it took me a while to figure it out and come up with a workaround. The following code will demonstrate the issue. Uncommenting the analysis snippet will resolve the issue.

ES version 6.3
ES DSL version 6.2.1

```python
from datetime import datetime
from elasticsearch import Elasticsearch
from elasticsearch_dsl import (
    analyzer,
    connections,
    Date,
    Document,
    Float,
    Index,
    Integer,
    Ip,
    Keyword,
    Mapping,
    Text,
    token_filter
)
from elasticsearch_dsl.utils import META_FIELDS
import json
import logging

logging.basicConfig(level=logging.DEBUG)

DocumentIndex = Index(
    'ingest-{}-{}'.format(
        "custom_doc",
        datetime.utcnow().isoformat()[:10].replace('-', '.')
    ),
    doc_type="custom_doc"
)


class BaseDocument(Document):

    timestamp = Date()
    size = Integer()
    version = Keyword()
    delayed_by = Float()
    indexed_at = Date()

    def __init__(self, meta=None, **kwargs):

        def do_sub(d):
            for k in d.keys():
                if k.startswith('_') and k[1:] in META_FIELDS:
                    continue
                v = d.pop(k)
                k = k.replace("-", "_").replace(".", "_")
                d[k] = v
                if isinstance(v, dict):
                    do_sub(v)
                elif isinstance(v, list):
                    for i in v:
                        if isinstance(i, dict):
                            do_sub(i)

        do_sub(kwargs)
        super(BaseDocument, self).__init__(meta=meta, **kwargs)


class FieldAnalyzers(object):

    email_filter = token_filter(
        "email_token_filter",
        type="pattern_capture",
        preserve_original=True,
        patterns=[
            "([^@]+)",
            "(\\p{L}+)",
            "(\\d+)",
            "@(.+)",
            "([^-@]+)"
        ]
    )
    email_analyzer = analyzer(
        "email_analyzer",
        tokenizer="uax_url_email",
        filter=[email_filter, "lowercase", "unique"],
        type="custom"
    )


@DocumentIndex.document
class CustomDoc(BaseDocument):

    class Meta:
        mapping = Mapping("custom_doc")

    class Index:
        doc_type = "custom_doc"
        name = DocumentIndex._name

    Username = Text(
        fields={
            "raw": Keyword(),
            "email": Text(analyzer=FieldAnalyzers.email_analyzer)
        }
    )
    IP_Address = Ip()
    Data = Text()
    Action = Keyword()


connections.add_connection("default", Elasticsearch("localhost"))

DocumentIndex.settings(
    **{"codec": "best_compression", "number_of_replicas": 1}
)
# analysis = DocumentIndex._mapping._collect_analysis()
# DocumentIndex.settings(
#     **{"analysis": analysis}
# )
DocumentIndex.aliases(**{"search-logs": {}})

index_tpl = DocumentIndex.as_template(
    'custom_doc_template',
    pattern="ingest-*"
)
print json.dumps(index_tpl.to_dict(), indent=4)
print json.dumps(DocumentIndex.to_dict(), indent=4)
index_tpl.save()
```

**Comment by azhard4int** [comment]
+1, just got started with templates and saw the same thing.

**Comment by HonzaKral** [comment]
Could you please verify that this wasn't fixed by cffaab4? Thanks!

**Comment by tstallings** [comment]
Nice, I'll try this as soon as I get home from work and let you know

**Comment by HonzaKral** [comment]
Assuming this was fixed in cffaab4, please feel free to reopen with additional information!
